golang/channel.go

```diff
@@ -140,10 +142,19 @@ func NewChannel(serviceName string, opts *ChannelOptions) (*Channel, error) {
 		connectionOptions: opts.DefaultConnectionOptions,
 		log:               logger.WithFields(LogField{"service", serviceName}),
 		statsReporter:     statsReporter,
-		traceReporter:     traceReporter,
 		handlers:          &handlerMap{},
 		subChannels:       &subChannelMap{},
 	}
+
+	traceReporter := opts.TraceReporter
+	if opts.TraceReporterFactory != nil {
```

Comment:
can you create a test to make sure that `TraceReporterFactory` is preferred over `TraceReporter`